pkg/healthz/healthz.go

```diff
@@ -0,0 +1,132 @@
+// Package healthz implements healthz handlers.
+package healthz
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net/http"
+	"sync"
+
+	"github.com/aws/aws-sdk-go/aws/request"
+	"go.uber.org/zap"
+	pb "k8s.io/apiserver/pkg/storage/value/encrypt/envelope/v1beta1"
+	"sigs.k8s.io/aws-encryption-provider/pkg/plugin"
+)
+
+// ErrHandler is returned when the handler is configured wrong.
+var ErrHandler = errors.New("'encrypt' checker must precede 'decrypt' checker or 'encrypt' checker not found")
+
+// NewHandler returns a new healthz handler.
+func NewHandler(srv *plugin.Plugin, cli pb.KeyManagementServiceClient, healthzCheckers []string) (http.Handler, error) {
+	encryptCkFound := false
+	for _, ck := range healthzCheckers {
+		_, ok := checkers[ck]
+		if !ok {
+			return nil, fmt.Errorf("checker %q unknown", ck)
+		}
+		if ck == "encrypt" {
+			encryptCkFound = true
+			continue
+		}
+		if ck == "decrypt" && !encryptCkFound { // "encrypt" must precede
+			return nil, ErrHandler
+		}
+	}
+	return &handler{
+		mu:              sync.RWMutex{},
+		srv:             srv,
+		cli:             cli,
+		healthzCheckers: healthzCheckers,
+	}, nil
+}
+
+type handler struct {
+	mu              sync.RWMutex
+	srv             *plugin.Plugin
+	cli             pb.KeyManagementServiceClient
+	healthzCheckers []string
+	cipher          []byte
+}
+
+func (hd *handler) ServeHTTP(rw http.ResponseWriter, req *http.Request) {
```

Comment:
I think this handler could probably be a little smarter, we should only call `kms:Encrypt` when there _hasn't_ been any encrypt/decrypt operations in the last _n_ seconds (_n_ could be hardcoded initially to 30 sec and we can always make that configurable later)